tools/flang2/flang2exe/dinit.cpp

```diff
@@ -4618,7 +4618,8 @@ mk_cmp(CONST *c, int op, INT l_conval, INT r_conval, DTYPE rdtype, DTYPE dt)
     break;
   case OP_XTOI:
   case OP_XTOK:
-    c->u1.conval = init_fold_const(get_ast_op(op), l_conval, r_conval, rdtype);   
```

Comment:
how could this possibly work without `break`? I hope you've tested it thoroughly

Reply:
The original code actually behaved correctly. I have rewritten the patch and added a test case.